content/2016.01.loadtesting.with-oauth-part-2.rst

```diff
@@ -0,0 +1,264 @@
+
+Then we can use this Bearer Token for our request:
+
+.. code-block:: bash
+
+    http https://profile.stage.mozaws.net/v1/profile \
+        Authorization:"Bearer b1d4babc01502fa8d9fc0139757168bf5908f33abc66f46253d4c69468e39373"
+
+
+Using fxa-client to create new test accounts
+--------------------------------------------
+
+We can also create a random user (on the stage environment only) and
+generate a Bearer Token for it:
+
+.. code-block:: bash
+
+    fxa-client --create --bearer --prefix my-app
+
+    # ---- BEARER TOKEN INFO ----
+    # User: my-app-6318a65dde1efc2f4c3f7b4e6cb33188@restmail.net
+    # Scopes: profile
+    # Account: https://api-accounts.stage.mozaws.net/v1
+    # Oauth: https://oauth.stage.mozaws.net/v1
+    # Client ID: 5882386c6d801776
+    # ---------------------------
+    export OAUTH_BEARER_TOKEN="90abc87ed1621ee504c1252ed382abc8269d1abc29f2ff87cc5e25f00249fabc"
+
+If we want to reuse the same account multiple time and create it only
+if needed, we can provide the user-salt (as a valid base64 string):
+
+.. code-block:: bash
+
+    fxa-client --create --bearer --prefix my-app --user-salt MySalt==
+
+
+Using fxa-client to work with BrowserID assertion
+-------------------------------------------------
+
+Even if we tend to switch from BrowserID assertion to Bearer Tokens
+for our new services, if we need to work with Firefox Sync or Firefox
+Hello we may need to provide a BrowserID assertion to log into these
+services.
+
+In that case ``fxa-client`` provides few specific attributes:
+
+.. code-block:: bash
+
+    optional arguments:
+      --browserid, --bid    Generate a BrowserID assertion
+      --audience AUDIENCE   Firefox BrowserID assertion audience.
+      --duration DURATION   Firefox BrowserID assertion duration.
+
+It works just like before with Bearer Tokens but generate a BrowserID
+assertion instead:
+
+.. code-block:: bash
+
+    fxa-client --create --browserid --prefix my-app --user-salt MySalt== \
+        --audience https://loop.stage.mozaws.net
+
+    # ---- BROWSER ID ASSERTION INFO ----
+    # User: my-app-b82d4afaf57cb856ccc04a58a07ce80f@restmail.net
+    # Audience: https://loop.stage.mozaws.net
+    # Account: https://api-accounts.stage.mozaws.net/v1
+    # ------------------------------------
+    export FXA_BROWSERID_ASSERTION="eyJhbGciOiJSUzI1NiJ9...hIQ9vrkqA"
+    export FXA_CLIENT_STATE="828aef3bc68ac0bde10f3d4b93303088"
+
+And then the assertion could be used like that:
+
+.. code-block:: bash
+
+    http POST https://loop.stage.mozaws.net/v0/registration \
+        Authorization:"BrowserID eyJhbGciOiJSUzI1NiJ9...hIQ9vrkqA"
+
+
+Using fxa-client to configure a loadtest
+========================================
+
+Now that we have a quite simple way to generate Bearer Tokens, how can
+we plug that with our load-tests?
+
+I could not find a better way than creating a bash file that export
+the environment variable and then loading it in the shell that will
```

Comment:
The end of the sentence " and then loading it in the shell that will run the loadtests" seems superfluous.